tests/test_modeling_utils.py

```diff
@@ -709,6 +712,126 @@ def test_infer_auto_device_map_with_buffer_check_and_multi_devices(self):
         assert len(w) == 0
         assert device_map == {"linear1": 0, "batchnorm": 1, "linear2": "cpu", "linear3": "cpu"}
 
+    def test_infer_auto_device_map_with_fallback_allocation(self):
+        # Create a model where modules cannot be allocated without fallback_allocation
+        # Define the inner module with its layers
+        inner_module = nn.Sequential(
+            OrderedDict([("linear1", nn.Linear(10, 4)), ("linear2", nn.Linear(4, 4)), ("linear3", nn.Linear(4, 8))])
+        )
+
+        # Wrap the inner module in another module
+        model = nn.Sequential(OrderedDict([("module", inner_module)]))
+
+        max_memory = {0: 256}
+
+        # Without fallback_allocation
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
Thanks for pointing that out! Should we check for the logs instead? 

I had a problem checking for cases where the message is not logged when there aren't any other logs since `with self.assertLogs() as cm` will raise ` AssertionError: no logs of level INFO or higher triggered on root`. Here is my workaround:
```
        try:
            with self.assertLogs() as cm:
                device_map = infer_auto_device_map(model, max_memory={0: 200, 1: 200})
                self.assertFalse(any("insufficient memory" in out for out in cm.output))
        except AssertionError:
            # No logs exist; test passes implicitly
            pass
```

Does this approach make sense when handling scenarios where logs might not be emitted? Let me know your thoughts!

Reply:
Sounds good to me ! Please go ahead =)

Reply:
Fixed. Thx!